### PR TITLE
NUTCH-2849 Replace remaining package.html files with package-info.java

### PR DESCRIPTION
**Pull request by lewismc**
PR to address https://issues.apache.org/jira/browse/NUTCH-2849
**failonwarning=true** flag now added... we have total Javadoc CI coverage.
There are a few package-info.java files missing but we can get around to those again at some stage in the future. 
Thanks @sebastian-nagel for the guidance on the last PR everything you said was correct.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -186,6 +186,7 @@
       doctitle="${name} ${version} API"
       bottom="Copyright &amp;copy; ${year} The Apache Software Foundation"
       failonerror="true"
+      failonwarning="true"
```
Comment (sebastian-nagel) [inline build.xml:189]: Observed two warnings when running `ant clean javadoc` (which made the build to fail):
``` 
  [javadoc] .../src/java/org/apache/nutch/protocol/RobotRulesParser.java:58: warning - Tag @link: reference not found: getRobotRulesSet
  [javadoc] .../src/plugin/indexer-elastic/src/java/org/apache/nutch/indexwriter/elastic/ElasticIndexWriter.java:204: warning - Tag @link: reference not found: BulkProcessor.Listener
  ...
  [javadoc] 2 warnings

BUILD FAILED
.../build.xml:723: Javadoc issued warnings.
```

File: build.xml [hunk 2]
```diff
@@ -809,6 +811,7 @@
       <classpath>
         <fileset dir="${build.plugins}" >
           <include name="**/*.jar"/>
+          <exclude name="any23/javax.annotation-api*.jar"/>
```
Comment (sebastian-nagel) [inline build.xml:814]: Ok. Nice, this solves the remaining "Multiple sources of package comments found" warning (see #568).

**Review by sebastian-nagel — COMMENTED** [review]
+1 looks good.

Are you able to reproduce the two warnings?

File: build.xml (hunk 2 above)
Comment (lewismc) [reply build.xml:814]: Hi @sebastian-nagel in short no I am not able to reproduce.
```% ant clean runtime
...
  [javadoc] Standard Doclet version 1.8.0_221
  [javadoc] Building tree for all the packages and classes...
  [javadoc] Building index for all the packages and classes...
  [javadoc] Building index for all classes...
     [copy] Copying 1 file to /Users/lmcgibbn/Downloads/nutch/build/docs/api/org/apache/nutch/plugin/doc-files
     [copy] Copying 1 file to /Users/lmcgibbn/Downloads/nutch/build/docs/api/resources
     [copy] Copying 1 file to /Users/lmcgibbn/Downloads/nutch/build/docs/api/resources

BUILD SUCCESSFUL
Total time: 38 seconds
```

File: build.xml (hunk 1 above)
Comment (lewismc) [reply build.xml:189]: Hi @sebastian-nagel in short no I am not able to reproduce.
```% ant clean runtime
...
  [javadoc] Standard Doclet version 1.8.0_221
  [javadoc] Building tree for all the packages and classes...
  [javadoc] Building index for all the packages and classes...
  [javadoc] Building index for all classes...
     [copy] Copying 1 file to /Users/lmcgibbn/Downloads/nutch/build/docs/api/org/apache/nutch/plugin/doc-files
     [copy] Copying 1 file to /Users/lmcgibbn/Downloads/nutch/build/docs/api/resources
     [copy] Copying 1 file to /Users/lmcgibbn/Downloads/nutch/build/docs/api/resources

BUILD SUCCESSFUL
Total time: 38 seconds
```

Comment (lewismc) [reply build.xml:189]: It looks like CI was not able to reproduce either...

**Pull request merged**